### ISSUE TITLE
StopWords doesn't seem to be working

### ISSUE THREAD

**Issue by unix-supremacist** [go-skynet/go-llama.cpp#183]
from reading [the doc](https://pkg.go.dev/github.com/go-skynet/go-llama.cpp#SetStopWords)

and reading the example `llama.SetStopWords("llama")` should stop predictions as soon as the prompt predicts this word correct? i can't get it to actually do that, it instead continues on.

i did momentarily dive into the code to see if i could understand what this was doing, to see if i just misunderstood the docs, or to see if i could figure out what was wrong, but i couldn't really find anything of note.

for now it can be faked with this code inserted to the example
```go
if strings.Contains(strings.ToLower(token), "llama"){
	return false
}
fmt.Print(token)
return true
```

**Comment by 65a** [comment]
I don't think this is a llama.cpp problem, because I can get `-r llama` to stop generation immediately after the word llama in interactive mode, however the reverse prompt == stop prompt feature seems poorly documented over there, and has some interaction with being in interactive mode or not. I can't figure out if this is a CGO problem converting a []string to a char**. Local effort to do this differently had the same result, so I think the next thing is to get llama.cpp to log whatever it thought the reverse prompt was. If it propagates, or doesn't we should be able to isolate the issue to one side of the interface between go-llama.cpp and the C++ side of things.

**Comment by 65a** [comment]
With some printfs added, I don't think the negative prompt (stopwords, for llama.cpp, which is confusing terminology) actually makes it to the binding.cpp
```
>>> What is an animal like an alpaca?

Sending What is an animal like an alpaca?

llama_predict: negative prompt is

llama_predict: negative prompt: ''
```
